src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -993,7 +998,8 @@ public static bool ParseDataSource(string dataSource, out string hostname, out i
                 {
                     return false;
                 }
-                dataSource = dataSource.Substring(0, dataSource.IndexOf(",", StringComparison.Ordinal) - 1);
+                // IndexOf is zero-based, no need to subtract one
+                dataSource = dataSource.Substring(0, dataSource.IndexOf(",", StringComparison.Ordinal));
```

Comment:
Good catch. Kind of surprised it hasn't failed earlier during testing. The dataSource should have been truncated by 1 character.

Reply:
It did actually fail during testing and found that the datasource was truncated.